### PR TITLE
feat: replace uniqueness multiplier with pioneer dividend mechanism

### LINKED ISSUE #240: Rewarding the pioneers
## Summary
Despite having 1400+ repos across all tiers, contribution activity clusters heavily on a small subset of popular repositories. When miners see others earning score on a repo, they follow

This creates two realities:

1. **Skewed contribution distribution**: A disproportionate share of contributions within each tier goes to a few top repos, defeating the purpose of having a broad repo list.
2. **Maintainer burden**: 'Popular' repos get flooded with PRs from miners chasing the same target. Maintainers are reviewing 10-20 PRs on the same repo.

The existing uniqueness multiplier attempts to address this, but the incentive isn't strong enough to change behavior (perhaps it needs to be removed and replaced with more ideal logic?).

## Motivation
Each PR earns a base token score from code changes, scaled by several multipliers (repo weight, issue bonus, time decay, credibility, spam penalty, and the repository uniqueness multiplier). The reward differential between a saturated repo and an untouched one isn't compelling enough to justify the effort of exploring a new codebase.

_Note: The scoring window only considers merged PRs within the last 90 days. This is relevant because "new to the network" for a repository should be evaluated against this same contribution window -- a repo that hasn't been contributed to within the scoring lookback is functionally 'undiscovered'._

## Scope
This issue involves:

- **Scoring logic**: The multiplier calculations and how they're applied during score finalization
- **Data model**: The relevant dataclass fields for any multipliers, scoring values, etc.
- **Database**: Any altered dataclass fields should stream down to their respective DB tables and columns
- **Storage layer**: Queries that insert/update/read any altered database columns
- **Tests**: Any existing tests should pass, plus new tests for the replacement mechanism
- **Logging**: Score finalization logs that display multiplier values & logs that display new scoring systems introduced

Don't leave orphaned columns, dead code, or untested logic.

## Task

- **Introduce a pioneer reward mechanism** that meaningfully incentivizes miners to be the first to contribute to repositories that have had no contributions within the scoring lookback window. The first miner to land a merged PR on an untouched repo should receive a substantial reward advantage compared to miners who follow them to that repo afterward. This should be calibrated and tuned to a fair value gain that is justifiable. 
- **The goal is not to punish followers** -- contributions to already-active repos should still score normally. The goal is to make pioneering genuinely attractive so that miners actively seek out new repos rather than piling onto proven ones.

## Preferences

- Be creative. A naive `if first_contributor: score *= 5` is not what we're looking for, think about the dynamics of the problem and take advantage of the opportunity to design something that handles them well.
- A clean, explainable mechanism beats a 10-variable equation that nobody can reason about. If you can't explain your approach in a few sentences, it's probably too complex.
- Code cleanliness and readability matter. Follow existing patterns, write code that reads naturally.
- Keep tests sharp and focused. Cover the important behavior and edge cases without overdoing it, don't pad test files for the sake of volume.

## PR Submission Requirements

When submitting a PR for this issue:

- **Solution explanation**: Concisely describe your approach and the reasoning behind your design choices. Why does your mechanism work better than the current one?
- **Evidence**: Demonstrate your solution works. Scoring simulation output showing before/after contribution distribution, test results, or whatever makes the strongest case. PRs without clear evidence of a working, tested solution will not be reviewed.
- **Full stack confirmation**: Confirm in your PR description that changes trace through every layer (dataclass, DB, queries, storage, scoring, constants, tests, logging).
- Follow all guidelines in `CONTRIBUTING.md`.
- All existing tests must pass. Add tests that cover the new mechanism's behavior, including edge cases (no prior contributions, repo just entered the list, multiple miners pioneering different repos in the same scoring cycle, etc.).

### PR DESCRIPTION
**Pull request by MkDev11**
## Summary

Replace the `repository_uniqueness_multiplier` with a **pioneer dividend** mechanism that rewards the first quality contributor to each repo with a share of followers' earned scores.

### Problem

Despite 1400+ repos across all tiers, contribution activity clusters on a small subset. The existing `repository_uniqueness_multiplier` gives a shared bonus (1.74x) that isn't compelling enough to change behavior — both pioneers and followers receive the same boost, creating no differential incentive to explore new repos.

### Solution

A pioneer dividend: the first miner to land a quality merged PR on an untouched repo earns a percentage of each follower's `earned_score` as an additive bonus. The dividend is per-PR, per-repo, and agnostic of who the miner is or what else they've pioneered.

```
raw_dividend     = follower_1.earned_score × 0.30
                 + follower_2.earned_score × 0.20
                 + sum(follower_3+.earned_score) × 0.10
pioneer_dividend = min(raw_dividend, pioneer_own_earned × 1.0)
earned_score     = base_score × repo_weight × issue × spam × decay × credibility + pioneer_dividend
```

### How it works

1. For each repo, collect all merged PRs that pass the quality gate (`token_score >= MIN_TOKEN_SCORE_FOR_BASE_SCORE`, valid tier config, non-null `merged_at`).
2. Per miner, keep only their earliest eligible PR on that repo.
3. Sort by `(merged_at, pr_number)` — position 1 is the pioneer.
4. Sum follower earned_scores weighted by position rates (30%/20%/10%).
5. Cap dividend at 1× pioneer's own `earned_score` (max 2× total).
6. Add dividend to pioneer's `earned_score` additively.

### Design decisions

- **Additive, not zero-sum**: Followers keep 100% of their score. The dividend is newly minted, not deducted. Per issue #240: *"The goal is not to punish followers."*
- **Per-PR, per-repo**: No multi-repo decay. Pioneering a second repo doesn't reduce the first repo's dividend. Per owner feedback: *"It makes more sense to me to reward the pioneers PR by PR, agnostic of who the miner is or what they've pioneered."*
- **Based on follower `earned_score`**: Dividend uses each follower's post-multiplier score. A follower with low credibility or a stale PR contributes less — the pioneer's reward reflects follower quality.
- **Capped at 1× own earned**: Prevents runaway dividends on popular repos. Self-calibrating — stale pioneers (low decay) have a lower cap.
- **`is_pioneer_eligible()`**: Instance method on `PullRequest` — the eligibility logic lives with the object it describes, per owner feedback.
- **3-phase `finalize_miner_scores`**: Phase 1 computes all `earned_scores` (`base × multipliers`). Phase 2 calculates dividends from follower earned_scores. Phase 3 aggregates totals + collateral.

### Anti-gaming protections

The mechanism is resistant to trivial-fix sniping (merging a small bug fix just to claim pioneer status):

1. **Quality gate**: `token_score >= MIN_TOKEN_SCORE_FOR_BASE_SCORE` — PRs with too few meaningful AST nodes are ineligible for pioneer status. One-liner typo fixes and README changes won't pass.
2. **Self-calibrating cap**: The dividend is capped at 1× the pioneer's own `earned_score`. A trivial fix earns a tiny `earned_score`, so even if 10 quality followers arrive, the sniper's dividend is capped at their own tiny score. Example: sniper merges a 0.5-point fix → max possible dividend = 0.5, regardless of follower activity.
3. **Time decay**: Stale pioneer PRs lose value over time via `time_decay_multiplier`, which reduces both the pioneer's `earned_score` and their dividend cap proportionally.
4. **Eligibility-scoped rank**: `pioneer_rank` is only set on PRs that pass `is_pioneer_eligible()`. Ineligible PRs from the same miner on the same repo keep the default rank of 0 — no false positives in the DB.

### Constants

| Constant | Value |
|----------|-------|
| `PIONEER_DIVIDEND_RATE_1ST` | 0.30 |
| `PIONEER_DIVIDEND_RATE_2ND` | 0.20 |
| `PIONEER_DIVIDEND_RATE_REST` | 0.10 |
| `PIONEER_DIVIDEND_MAX_RATIO` | 1.0 |

### Scoring examples (from real PRs)

> In all examples, `Repo Unique` multiplier is **removed** (replaced by pioneer dividend).
> `earned_score = base × repo_weight × issue × cred × decay` (no repo_unique).

#### Example 1: (Bronze) — 2 followers on same repo

| PR | base | repo_w | issue | cred | decay | old score | new earned |
|----|------|--------|-------|------|-------|-----------|------------|
| PR 1 (follower) | 14.04 | 0.89 | 1.75 | 0.97 | 0.96 | 28.32 | 20.36 |
| PR 2 (follower) | 11.74 | 0.89 | 1.75 | 0.97 | 0.82 | 20.22 | 14.54 |

Pioneer's dividend from these 2 followers:
```
dividend = 20.36 × 0.30 + 14.54 × 0.20 = 6.11 + 2.91 = 9.02
```
If pioneer's own earned = 20.00, then: **20.00 + 9.02 = 29.02**

#### Example 2: (Gold) — high repo weight follower

| PR | base | repo_w | issue | cred | decay | old score | new earned |
|----|------|--------|-------|------|-------|-----------|------------|
| PR 1 (follower) | 8.33 | 14.36 | 1.75 | 0.65 | 0.86 | 191.58 | 116.82 |

Pioneer's dividend from 1 follower:
```
dividend = 116.82 × 0.30 = 35.05
```
Gold-tier PRs with high repo weight generate large dividends — pioneering a popular Gold repo is very rewarding.

#### Example 3: (Gold) — low credibility follower

| PR | base | repo_w | issue | cred | decay | old score | new earned |
|----|------|--------|-------|------|-------|-----------|------------|
| PR 1 (follower) | 14.31 | 20.88 | 1.00 | 0.58 | 1.00 | 297.98 | 173.30 |

Pioneer's dividend from 1 follower:
```
dividend = 173.30 × 0.30 = 51.99
```
Despite cred=0.58 reducing the follower's earned_score, the high repo_weight (20.88x) still produces a large dividend. If the follower had cred=1.0, earned would be 298.77 → dividend = **89.63**.

#### Example 4: Cross-tier comparison

| Repo | Tier | Follower earned | 1st follower dividend |
|------|------|-----------------|-----------------------|
| Repo 1 | Bronze | 20.36 | 6.11 |
| Repo 2 | Silver | 68.78 | 20.63 |
| Repo 3 | Gold | 116.82 | 35.05 |

Higher-tier repos naturally produce larger dividends because `repo_weight` is larger. Pioneering a Gold repo pays ~6x more than pioneering a Bronze repo.

#### Example 5: Stale pioneer (high decay)

Pioneer merged early but decay=0.30, cred=0.97, base=14.04:
```
pioneer_earned = 14.04 × 0.89 × 1.75 × 0.97 × 0.30 = 6.37
dividend       = 20.36 × 0.30 + 14.54 × 0.20 = 9.02
cap            = min(9.02, 6.37 × 1.0) = 6.37
pioneer_total  = 6.37 + 6.37 = 12.74
```
The pioneer's own decay (0.30) crushes their base earned **and** their cap — but the raw dividend calculation is unaffected because it uses the **followers'** multipliers, not the pioneer's.

## Related Issues

Closes #240

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Other (describe below)

### Scoring scenario: 5 miners, 1 repo (pile-on)

| Miner | Role | earned_score | Dividend | Total |
|-------|------|-------------|----------|-------|
| UID 1 | Pioneer | 30.0 | 21.0 | 51.0 |
| UID 2 | Follower (1st) | 30.0 | 0.0 | 30.0 |
| UID 3 | Follower (2nd) | 30.0 | 0.0 | 30.0 |
| UID 4 | Follower (3rd) | 30.0 | 0.0 | 30.0 |
| UID 5 | Follower (4th) | 30.0 | 0.0 | 30.0 |

Pioneer advantage: **70%** over followers. Old uniqueness multiplier gave **0%** differential (everyone got 1.74x).

### Test results

```
$ python3 -m ruff check <all 8 changed files>
All checks passed!

$ python3 -m pytest tests/ -v
331 passed in 1.22s (0 failures, 0 regressions)
```

22 pioneer dividend tests across 4 classes:
- **TestPioneerEligibility** (5): Quality gate, tier requirement, merge timestamp
- **TestCalculatePioneerDividends** (13): Solo pioneer, follower dividend, multiple followers, cap, multi-PR summing, repo independence, quality exclusion, ineligible rank guard, tie-breaking, follow-up PRs, edge cases
- **TestFinalizeWithDividend** (2): Formula verification, follower score preservation
- **TestPioneerIncentiveEvidence** (2): Pile-on vs exploration, self-scaling reward

## Full Stack Confirmation

| Layer | File | Change |
|-------|------|--------|
| Constants | `gittensor/constants.py` | `PIONEER_DIVIDEND_RATE_{1ST,2ND,REST}`, `PIONEER_DIVIDEND_MAX_RATIO`; removed `UNIQUE_PR_BOOST` |
| Dataclass | `gittensor/classes.py` | `pioneer_dividend`, `pioneer_rank` fields + `is_pioneer_eligible()` method |
| Scoring | `gittensor/validator/evaluation/scoring.py` | `calculate_pioneer_dividends()` + 3-phase `finalize_miner_scores()`; removed `calculate_uniqueness_multiplier()` |
| Logging | `gittensor/validator/evaluation/reward.py` | Updated comment |
| DB/Queries | `gittensor/validator/storage/queries.py` | `pioneer_dividend` + `pioneer_rank` columns replace `repository_uniqueness_multiplier` |
| Storage | `gittensor/validator/storage/repository.py` | Updated storage tuple |
| Test fixtures | `tests/validator/conftest.py` | Added `uid` + `merged_at` params to `PRBuilder` |
| Tests | `tests/validator/test_pioneer_dividend.py` | 21 tests across 4 classes |

No orphaned columns, dead code, or untested logic.

## Testing

- [x] Tests added/updated
- [x] Manually tested

## Checklist

- [x] Code follows project style guidelines
- [x] Self-review completed
- [x] Changes are documented (if applicable)

### CODE REVIEW

File: gittensor/validator/evaluation/scoring.py [hunk 1]
```diff
@@ -304,13 +307,93 @@ def calculate_time_decay_multiplier(pr: PullRequest) -> float:
     return max(sigmoid, TIME_DECAY_MIN_MULTIPLIER)
 
 
+def calculate_pioneer_dividends(
```
Comment (LandynDev) [inline gittensor/validator/evaluation/scoring.py]: This function structurally makes sense and looks good. However, I'm counting 9 for loops or something in this function. Are we sure there is no way to simplify this implementation? Do you mind reviewing this with AI and seeing if there's simplification opportunities while retaining same logic?

**Review by LandynDev — APPROVED** [review]
